### ISSUE TITLE
Publishable library won't build (out of the box Nx install)

### ISSUE THREAD

**Issue by alindsay55661** [nrwl/nx#6739]
## Current Behavior

Out of the box publishable components are not working. 

```
npx create-nx-workspace --preset=next
```

```
nx g @nrwl/next:library ui-components --directory=shared --publishable --importPath=components
nx build shared-ui-components --verbose
```

## Expected Behavior

The library generated by Nx should compile correctly.

## Steps to Reproduce

This issue is reproduced in the following repo.
https://github.com/alindsay55661/nx-error

Simply checkout the repo and run:

```
nx build shared-ui-components --verbose
```

### Failure Logs

```
> nx run shared-ui-components:build --verbose 
Bundling shared-ui-components...
Error during bundle: Error: You have declared using "bundled" babelHelpers, but transforming C:\Users\alanlindsay\projects\nx-error\libs\shared\ui-components\src\index.ts resulted in "runtime". Please check your configuration.
Bundle failed: shared-ui-components
Error during bundle: Error: You have declared using "bundled" babelHelpers, but transforming C:\Users\alanlindsay\projects\nx-error\libs\shared\ui-components\src\index.ts resulted in "runtime". Please check your configuration.
Bundle failed: shared-ui-components

———————————————————————————————————————————————

>  NX   ERROR  Running target "shared-ui-components:build" failed

  Failed tasks:

  - shared-ui-components:build

  Hint: run the command with --verbose for more details.
```

### Environment

```
>  NX  Report complete - copy this into the issue template

  Node : 14.15.5
  OS   : win32 x64
  npm  : 6.14.11

  nx : Not Found
  @nrwl/angular : Not Found
  @nrwl/cli : 12.7.1
  @nrwl/cypress : 12.7.1
  @nrwl/devkit : 12.7.1
  @nrwl/eslint-plugin-nx : 12.7.1
  @nrwl/express : Not Found
  @nrwl/jest : 12.7.1
  @nrwl/linter : 12.7.1
  @nrwl/nest : Not Found
  @nrwl/next : 12.7.1
  @nrwl/node : Not Found
  @nrwl/nx-cloud : Not Found
  @nrwl/react : 12.7.1
  @nrwl/schematics : Not Found
  @nrwl/tao : 12.7.1
  @nrwl/web : 12.7.1
  @nrwl/workspace : 12.7.1
  @nrwl/storybook : 12.7.1
  @nrwl/gatsby : Not Found
  typescript : 4.3.5
```

**Comment by nevestuan** [comment]
I face the same issue. There is something wrong with the .babelrc configuration. I managed to make it work by changing the presets configuration as below:

from: "presets": ["next/babel"]
to: "presets": [
    [
      "@nrwl/react/babel",
      {
        "runtime": "automatic",
        "useBuiltIns": "usage"
      }
    ]
  ]

**Comment by jaysoo** [comment]
The fix for this is coming for 12.10.0, which is scheduled for next week (10/5).

**Comment by jaysoo** [comment]
or now you can switch the babelrc as @nevestuan outlined. Sorry for the inconvenience!
